packages/runtime/src/next/client.ts

```diff
@@ -201,23 +230,51 @@ export class Makeswift {
     return response
   }
 
-  async getPages({
-    siteVersion = MakeswiftSiteVersion.Live,
-  }: {
-    siteVersion?: MakeswiftSiteVersion
-  } = {}): Promise<MakeswiftPage[]> {
-    const response = await this.fetch(`v3/pages`, siteVersion)
+  private getPagesInternal = async (
+    args: {
+      siteVersion?: MakeswiftSiteVersion
+      limit?: number
+      after?: string
+      sortBy?: 'title' | 'path' | 'createdAt' | 'updatedAt' | 'publishedAt'
+      sortDirection?: 'asc' | 'desc'
+      includeOffline?: boolean
+      pathPrefix?: string
+      locale?: string
+    } = {},
+  ) => {
+    const {
+      siteVersion = MakeswiftSiteVersion.Live,
+      limit = 20,
+      after,
+      includeOffline,
+      sortBy,
+      sortDirection,
+      pathPrefix,
+      locale,
+    } = args
+
+    const url = new URL('v4/pages', this.apiOrigin)
+    url.searchParams.set('limit', limit.toString())
+    if (after != null) url.searchParams.set('after', after)
+    if (includeOffline != null) url.searchParams.set('includeOffline', String(includeOffline))
+    if (sortBy != null) url.searchParams.set('sortBy', sortBy)
+    if (sortDirection != null) url.searchParams.set('sortDirection', sortDirection)
+    if (pathPrefix != null) url.searchParams.set('pathPrefix', pathPrefix)
+    if (locale != null) url.searchParams.set('locale', locale)
```

Comment:
I'd consider

```typescript
const query = getPagesQueryParams(args)
const response = await this.fetch('v4/pages?' + query, args.siteVersion)
```

where `getPagesQueryParams`  takes the args and returns `URLSearchParams`.

Reply:
Note that I'm not suggesting factoring out `getPagesQueryParams` for reuse in any other place, but rather as a way to "tidying up" the implementation here. Basically we have 20 lines of code that are dedicated to constructing the query params, inlined and mixed in with the much more compact and higher-level code that does the work of calling the endpoint and transforming the result.

Reply:
Here are a couple of links on the topic: 
* https://medium.com/@brooknovak/single-level-of-abstraction-d7fe8aa5c813
* https://dev.to/bornfightcompany/why-you-shouldn-t-mix-levels-of-abstraction-in-your-functions-332f

Reply:
Actually, maybe Zod can take care of this for us? Is it possible for it to exclude `undefined` keys?

```ts
new URLSearchParams(serachParamsSchema.parse(params))
```

Reply:
Resolved, moved param construction outside the client class.